build.py

```diff
@@ -45,50 +51,61 @@
 # Convert encoding before compiling.
 # asmproc_flags += ["--input-enc", "utf-8", "--output-enc", "euc-jp"]
 
-preprocessed_file = tempfile.NamedTemporaryFile(
-    prefix="preprocessed", suffix=".c", delete=False
-)
+with tempfile.TemporaryDirectory(prefix="asm_processor") as tmpdirname:
+    tmpdir_path = Path(tmpdirname)
+    preprocessed_path = tmpdir_path / "preprocessed.c"
```

Comment:
I have vague memories about IDO using the basename for temp files, causing this use of identical names to result in possible make -j breakage. I can't reproduce it now in testing though, other than on -O3 which asm-processor doesn't support anyway. Could be worth randomizing it, still...

Reply:
.o files aren't worrisome because we only ever compile with -o and specify the output destination directly. Using a uuid sounds good!